pkg/query/filter/request.go

```diff
@@ -18,17 +32,41 @@ type RequestOption struct {
 	MultiSelect bool
 }
 
+// ReadableValue is a generic type that represents a human-readable value with a corresponding backend value.
+// It has two fields: `Label` (the human-readable form of the value) and `Value` (the value for the backend).
+//
+// Usage example:
+//
+//	type RequestOption struct {
+//	    Name        ReadableValue[string]
+//	    Control     RequestOptionType
+//	    Operators   []ReadableValue[CompareOperator]
+//	    Values      []string
+//	    MultiSelect bool
+//	}
 type ReadableValue[T any] struct {
 	// Label is the human-readable form of the value
 	Label string `json:"label"`
 	// Value is the value for the backend
 	Value T `json:"value"`
 }
 
+// RequestOptionType configures the type of control for a field in a request option.
+// It has a `Type` field which is of type `ControlType`.
```

Comment:
Do we really want to document it in such a detailled way? If the implementation only changes a bit, we must also remember to change the comment :-D
For me the first line would be totally sufficient!